src/fs/filesystem.ts

```diff
@@ -145,6 +146,28 @@ export class FileSystem {
     })
   }
 
+  async mv(from: string, to: string): Promise<CID> {
+    const node = await this.get(from)
+    if (node === null) {
+      throw new Error(`Path does not exist: ${from}`)
+    } 
+    const toParts = pathUtil.splitParts(to)
+    const destPath = pathUtil.join(toParts.slice(0, toParts.length - 2)) // remove file/dir name
```

Comment:
I think this should be `toParts.length - 1`, because the `end` (aka. 2nd param) is non-inclusive.
See docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice#Parameters

Reply:
yup you're totally right! I actually just stumbled on that myself as I was testing :+1: